core/embed/extmod/modtrezorutils/modtrezorutils-meminfo.h

```diff
@@ -253,17 +253,13 @@ void dump_dict_inner(FILE *out, const mp_obj_dict_t *dict) {
 
 void dump_function(FILE *out, const mp_obj_fun_bc_t *func) {
   print_type(out, "function", NULL, func, false);
-  fprintf(out, ",\n\"globals\": \"%p\"", func->globals);
+  fprintf(out, ",\n\"globals\": \"%p\"", func->context->module.globals);
```

Comment:
what kind of thing is `context`? if it's something that the meminfo can understand, we should be dumping the `context` instead of going through to globals

Reply:
Right, [context](https://github.com/micropython/micropython/blob/v1.19.1/py/bc.h#L209-L213) is module + constants. We can dump the module, [constants](https://github.com/micropython/micropython/blob/v1.19.1/py/bc.h#L199-L207) are in ROM I think and therefore not interesting for us?
c29a1a88a7fd312f750274e8934e299150125fe7

Reply:
Like 34af8044ed1b01da0f6a14490723c1e5307bdbc7? We could also pass `func->context` to `dump_value` because it has `mp_obj_module_t` at the beginning but this seems more explicit.

Reply:
i think we should _also_ pass the context to `dump_value`, and add support for the context type.
because this way you get the pointer but not content of the pointee -- in the rendered memory dump, you will not see how functions hold a reference to their module -- just a link to an opaque pointer, and then then module separately